packages/hawtio/src/plugins/camel/CamelContent.tsx

```diff
@@ -209,14 +209,21 @@ const CamelContentContextToolbar: React.FunctionComponent = () => {
       const attr = await contextsService.getContext(selectedNode)
       if (attr) setContextState(attr)
 
-      contextsService.register({ type: 'read', mbean: objectName }, (response: Response) => {
-        log.debug('Scheduler - Contexts:', response.value)
-
-        // Replace the context in the existing set with the new one
-        const attrs = response.value as AttributeValues
-        const newAttrs = contextsService.toContextState(selectedNode, attrs)
-        if (newAttrs) setContextState(newAttrs)
-      })
+      contextsService.register(
+        { type: 'read', mbean: objectName },
+        (response: JolokiaSuccessResponse | JolokiaErrorResponse) => {
+          if (Jolokia.isError(response)) {
```

Comment:
Not related to this pull req, but ideally `Jolokia.isError` should then narrow down the type of response to `JolokiaErrorResponse` with `response is JolokiaErrorResponse` at return so that you don't need to cast the type again inside the if statement.
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
Since this type guard is used in many places, we can declare some custom `isJolokiaError` type guard function in `util/jolokia.ts`, and we'll no longer need this type casting for both error and success types.

We may also declare a type alias to compact the argument definitions:
```
type JolokiaResponse = JolokiaSuccessResponse | JolokiaErrorResponse
```

Reply:
I really tried the type guard (`response is ...`), but somehow it didn't work... In my simple scenario I understood how it works and tells TS about the types in `if` and `else` branches.
But somehow it didn't work with Jolokia types.

But now I checked in this particular file and it worked... ;)

Reply:
I made `Jolokia.isError` an actual type predicate here: https://github.com/jolokia/jolokia/commit/29eeeafab8e7ddf838c7055afcbd84ef509245a5